Frontend-project/14-age-calculator/scripts/script.js

```diff
@@ -0,0 +1,48 @@
+let userInput = document.getElementById('date');
+userInput.max = new Date().toISOString().split('T')[0];
+let result = document.getElementById('result');
+
+function calculateAge() {
+    let birthDate = new Date(userInput.value);
+
+    let d1 = birthDate.getDate();
+    let m1 = birthDate.getMonth() +1;
+    let y1 = birthDate.getFullYear();
+
+    let today = new Date();
+
+    let d2 = today.getDate();
+    let m2 = today.getMonth() +1;
+    let y2 = today.getFullYear();
+
+    let d3, m3, y3;
+
+    y3 = y2-y1;
+
+    if (m2 >= m1) {
+        m3 = m2-m1;
+    }
+    else{
+        y3--;
+        m3 = 12 + m2 - m1;
+    }
+
+    if (d2 >= d1) {
+        d3 = d2-d1;    
+    }
+    else{
+        m3--;
+        d3 = getDaysInMonth(y1, m1) + d2 - d1;
+    } 
+    
+    if (m3 < 0) {
+        m3 = 11;
+        y3--;
+    }
+    result.innerHTML = `You are ${d3} days, ${m3} months and ${y3} years`;
+}
```

Comment:
This `calculateAge` function looks too complex and the variable namings are not understandable. Go through again and refactor with a much simpler approach and presentation